java/client/src/test/java/glide/connection/ConnectionWithGlideMockTests.java

```diff
@@ -0,0 +1,203 @@
+        RustCoreMock.updateGlideMock(
+                new RustCoreMock.GlideMockProtobuf() {
+                    @Override
+                    public Response connection(ConnectionRequest request) {
+                        return null;
+                    }
+
+                    @Override
+                    public Response.Builder redisRequest(RedisRequest request) {
+                        return null;
+                    }
+                });
+
+        assertThrows(TimeoutException.class, () -> testConnection().get(1, SECONDS));
+    }
+
+    @Test
+    @SneakyThrows
+    public void cant_connect_when_negative_response() {
+        RustCoreMock.updateGlideMock(
+                new RustCoreMock.GlideMockProtobuf() {
+                    @Override
+                    public Response connection(ConnectionRequest request) {
+                        return Response.newBuilder().setClosingError("You shall not pass!").build();
+                    }
+
+                    @Override
+                    public Response.Builder redisRequest(RedisRequest request) {
+                        return null;
+                    }
+                });
+
+        var exception = assertThrows(ExecutionException.class, () -> testConnection().get(1, SECONDS));
+        assertAll(
+                () -> assertTrue(exception.getCause() instanceof ClosingException),
+                () -> assertEquals("You shall not pass!", exception.getCause().getMessage()));
+    }
+
+    @Test
+    @SneakyThrows
+    public void rethrow_error_on_read_when_malformed_packet_received() {
+        RustCoreMock.updateGlideMock(
+                new RustCoreMock.GlideMock() {
+
+                    @Override
+                    public byte[] handle(byte[] request) {
+                        return new byte[] {-1};
+                    }
+                });
+
+        var exception = assertThrows(ExecutionException.class, () -> testConnection().get(1, SECONDS));
+        assertAll(
+                () -> assertTrue(exception.getCause() instanceof ClosingException),
+                () ->
+                        assertTrue(
+                                exception
+                                        .getCause()
+                                        .getMessage()
+                                        .contains("An unhandled error while reading from UDS channel")));
+    }
+
+    @Test
+    @SneakyThrows
+    public void rethrow_error_if_UDS_channel_closed() {
+        var client = new TestClient(channelHandler);
+        stopRustCoreLibMock();
+        try {
+            var exception =
+                    assertThrows(
+                            ExecutionException.class, () -> client.customCommand(new String[0]).get(1, SECONDS));
+            assertTrue(exception.getCause() instanceof RuntimeException);
+
+            // Not a public class, can't import
+            assertEquals(
+                    "io.netty.channel.StacklessClosedChannelException",
```

Comment:
This seems brittle if Netty's version changes.

Reply:
hopefully they make it public when they update?

Reply:
I hope it won't happen without CI verification to catch failures.